fvm/src/call_manager/default.rs

```diff
@@ -385,9 +403,19 @@ where
             )?;
 
         log::trace!("calling {} -> {}::{}", from, to, method);
+        let gas_limit = self.gas_tracker.gas_limit().round_down() as u64;
         self.map_mut(|cm| {
             // Make the kernel.
-            let kernel = K::new(cm, block_registry, from, to, method, value.clone());
+            let kernel = K::new(
+                cm,
+                block_registry,
+                from,
+                to,
+                method,
+                value.clone(),
+                gas_premium.clone(),
+                gas_limit,
```

Comment:
Doesn't the Kernel already have access to the `gas_limit` via the `call_manager.gas_tracker`? Why does it need to be handed off the field explicitly? (for premium, see above)

Reply:
oh, didn't notice that -- yeah no need for it then.

Reply:
done.